lib/schemes/refresh.js

```diff
@@ -0,0 +1,329 @@
+      endpoint,
+      this.options.endpoints.refresh
+    ).then(response => {
+      this._updateTokens(response)
+    }).catch(() => {
+      // TODO: Unhandled error
+      this._logoutLocally()
+    })
+  }
+
+  _scheduleTokenRefresh () {
+    // If auto refresh is disabled, bail
+    if (this.options.disableAutoRefresh) return
+
+    let intervalDuration = (this._getExpiration() - Date.now()) * 0.75
+    if (intervalDuration < 1000) {
+      // in case you misconfigured refreshing this will save your auth-server from a self-induced DDoS-Attack
+      intervalDuration = 1000
+    }
+
+    this.refreshInterval = setInterval(() => {
+      this._tokenRefresh()
+    }, intervalDuration)
+  }
+
+  _initializeErrorInterceptor () {
+    this.$auth.ctx.app.$axios.onError(error => {
+      // If error has config and response and error status is 401
+      if (error.config && error.response && error.response.status === 401) {
+        const token = this.$auth.getToken(this.name)
+        const headerToken = error.config.headers[this.options.tokenName]
+
+        // If current token is equal to header token
+        if (token && headerToken && headerToken === token) {
+          // Add the failed request config to failedRequestQueue
+          if (!error.config.__isRetryRequest) {
+            error.config.__isRetryRequest = true
+            this.failedRequestQueue.push(error.config)
+          }
+
+          // If is not refreshing the token
+          if (!this.isRefreshing) {
+            // Refresh it
+            return this._tokenRefresh().then(() => {
+              this._syncExpiration()
+              this._scheduleTokenRefresh()
+            })
+          } else {
+            // Or resolve Promise
+            return Promise.resolve()
+          }
+        } else if (token && headerToken && headerToken !== token && !error.config.__isRetryRequest) {
+          // Or just retry the request if token is not equal to header token
+          error.config.__isRetryRequest = true
+          error.config.headers[this.options.tokenName] = token
+          return this.$auth.ctx.app.$axios.request(error.config)
+        }
+      }
+
+      // Otherwise, reject Promise
+      return Promise.reject(error)
+    })
+  }
+
+  async mounted () {
+    if (this.options.tokenRequired) {
+      const token = this.$auth.syncToken(this.name)
+      this._setToken(token)
+      this._syncRefreshToken()
+
+      const expiration = this._syncExpiration()
+      if (Date.now() >= expiration) {
+        await this._logoutLocally()
+      }
+
+      this.$auth.syncClientId(this.name)
+    }
+
+    return this.$auth.fetchUserOnce().then((response) => {
+      // Initialize axios error interceptor
+      this._initializeErrorInterceptor()
+
+      // Only refresh token if user is logged in and is client side
+      if (process.client && this.$auth.loggedIn) {
+        setTimeout(() => {
+          this._tokenRefresh().then(() => {
+            this._syncExpiration()
```

Comment:
Why are we not syncing expiration within the tokenRefresh function?

Reply:
I guess you are right, we could remove it.